extmod/vfs_posix_file.c

```diff
@@ -93,6 +93,7 @@ mp_obj_t mp_vfs_posix_file_open(const mp_obj_type_t *type, mp_obj_t file_in, mp_
     }
 
     o->base.type = type;
+    o->fd = -1;
```

Comment:
`mp_obj_str_get_str()` above could throw, so this probably needs to be before that (right after `m_new_obj_with_finalise()`.

Reply:
It probably wouldn't hurt to add a comment too so that no one tries to remove it or move it in the future.

Reply:
Thanks, will update.

Reply:
OK, I've moved the assignment up. Any further refinements to that code are above my current understanding of MicroPython's internals, so I'll let somebody else handle that.